File: app/src/main/java/me/imzack/app/ender/view/contract/TypeSearchViewContract.java
package me.imzack.app.ender.view.contract;

import me.imzack.app.ender.view.adapter.TypeSearchListAdapter;

public interface TypeSearchViewContract extends BaseViewContract {

    void showInitialView(int typeCount, TypeSearchListAdapter typeSearchListAdapter);

    void onSearchChanged(boolean isNoSearchInput, boolean isTypeSearchEmpty);

    void onTypeItemClicked(int typeListPos);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/TypeSearchViewContract.kt
package net.zackzhang.app.end.view.contract

import net.zackzhang.app.end.view.adapter.TypeSearchListAdapter

interface TypeSearchViewContract : BaseViewContract {

    fun showInitialView(typeCount: Int, typeSearchListAdapter: TypeSearchListAdapter)

    fun onSearchChanged(isNoSearchInput: Boolean, isTypeSearchEmpty: Boolean)

    fun onTypeItemClicked(typeListPos: Int)
}
